### PR TITLE
Scenes: Compare versions in dashboard settings

### PR DESCRIPTION
**Pull request by mdvictor**
Covers the third part - "Two versions from the list can be selected and compared" - of versions settings page migration https://github.com/grafana/grafana/issues/80058

### CODE REVIEW

File: public/app/features/dashboard-scene/settings/VersionsEditView.tsx [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 
 import { DashboardEditView, DashboardEditViewState, useDashboardEditPageNav } from './utils';
-import { RevisionsModel, VersionHistoryTable, historySrv } from './version-history';
+import {
+  RevisionsModel,
+  VersionHistoryComparison,
+  VersionHistoryHeader,
+  VersionHistoryTable,
+  VersionsHistoryButtons,
+  historySrv,
```
Comment (dprokop) [inline public/app/features/dashboard-scene/settings/VersionsEditView.tsx:19]: @mdvictor - I see these being duplicated in the old runtime directory and here. Think we should only keep the version in `dashboard-scene` dir, and get rid of the old runtime versions of these files. 

My point was - let's move what's possible to move without a major refactor under the `dashboard-scene` feature dir, and reference those in the old architecture imports.

**Review by dprokop — COMMENTED** [review]
Looking and working good!

File: public/app/features/dashboard-scene/settings/VersionsEditView.tsx (hunk 1 above)
Comment (mdvictor) [reply public/app/features/dashboard-scene/settings/VersionsEditView.tsx:18]: I migrated these files and left the others in the old `VersionHistory` folder for now, they will be touched in the restoration PR.

**Comment by mdvictor** [comment]
This PR grew a bit big, I ended up migrating a bunch more files and doing small refactors - mostly around styles needing to be represented as objects - and changed the imports in the old arch. I left the restore modal and logic in the old arch and will probably need to reimplement some parts of those for scenes. So I will copy those in scenes, refactor and keep 2 versions one for new and one for the old arch.

**Review by dprokop — APPROVED** [review]
Nice, looking good!

**Pull request merged**